benchmark/checks/utils.py

```diff
@@ -0,0 +1,186 @@
+from aea.mail.base import Envelope
+from aea.protocols.base import Protocol
+from aea.protocols.default.message import DefaultMessage
+from aea.registries.resources import Resources
+from aea.skills.base import Skill, SkillContext
+
+ROOT_DIR = os.path.join(os.path.dirname(inspect.getfile(inspect.currentframe())))  # type: ignore
+
+
+def wait_for_condition(condition_checker, timeout=2, error_msg="Timeout") -> None:
+    """Wait for condition occures in selected timeout."""
+    start_time = time.time()
+
+    while not condition_checker():
+        time.sleep(0.0001)
+        if time.time() > start_time + timeout:
+            raise TimeoutError(error_msg)
+
+
+def make_agent(runtime_mode="threaded") -> AEA:
+    """Make AEA instance."""
+    agent_name = "my_agent"
+    private_key_path = os.path.join(ROOT_DIR, DEFAULT_PRIVATE_KEY_FILE)
+    wallet = Wallet({DEFAULT_LEDGER: private_key_path})
+    identity = Identity(agent_name, address=wallet.addresses[DEFAULT_LEDGER])
+    resources = Resources()
+    resources.add_skill(
+        Skill.from_dir(
+            str(Path(AEA_DIR, "skills", DEFAULT_SKILL.name)), agent_context=MagicMock()
+        )
+    )
+    resources.add_protocol(
+        Protocol.from_dir(str(Path(AEA_DIR, "protocols", DEFAULT_PROTOCOL.name)))
+    )
+    return AEA(identity, wallet, resources, runtime_mode=runtime_mode)
+
+
+def make_envelope(sender: str, to: str) -> Envelope:
+    """Make an envelope."""
+    message = DefaultMessage(
+        dialogue_reference=("", ""),
+        message_id=1,
+        target=0,
+        performative=DefaultMessage.Performative.BYTES,
+        content=b"content",
+    )
+    message.sender = sender
+    message.to = to
+    return Envelope(
+        to=to, sender=sender, protocol_id=DefaultMessage.protocol_id, message=message,
+    )
+
+
+class GeneratorConnection(Connection):
+    """Generates messages and count."""
+
+    connection_id = PublicId("fetchai", "generator", "0.1.0")
+
+    def __init__(self, *args, **kwargs):
+        """Init connection."""
+        super().__init__(*args, **kwargs)
+        self._enabled = False
+        self._count_in = 0
+        self._count_out = 0
+
+    def enable(self) -> None:
+        """Enable message generation."""
+        self._enabled = True
+
+    def disable(self) -> None:
+        """Disable message generation."""
+        self._enabled = False
+
+    async def connect(self) -> None:
+        """Connect connection."""
+        self._state.set(ConnectionStates.connected)
+
+    async def disconnect(self):
+        """Disonnect connection."""
+        self._state.set(ConnectionStates.disconnected)
+
+    async def send(self, envelope: "Envelope") -> None:
+        """Handle incoming envelope."""
+        self._count_in += 1
+
+    async def receive(self, *args, **kwargs) -> Optional["Envelope"]:
+        """Generate an envelope."""
+        while not self._enabled:
+            await asyncio.sleep(0.0001)
```

Comment:
declare value as constant, no magic values.

Also what is the "right" value here?

Reply:
there is no right value, it's done only for specific testing purposes